src/hmt-client-processor.js

```diff
@@ -312,61 +312,47 @@ var hmt_client_processor = function(settings){
     })
 
     // handle any processing errors here
-    if(!res || !res.isSuccessful){
-
-      this._add_internal_error('Fullsteam, Could not get token')
-
-      var msg = ''
-
-      if(res && res.responseDetails){
-        for(var key in res.responseDetails)
-          this._add_processing_error(res.responseDetails[key].message)
-      }
-
-      if(res && res.issuerResponseDetails){
-
-        var issuerResponseCode = res.issuerResponseDetails.issuerResponseCode || 0
-        var issuerResponseDescription = res.issuerResponseDetails.issuerResponseDescription || ''
-        var CVVResponseCode = res.issuerResponseDetails.cvvResponseCode && ['M', 'P'].indexOf(res.issuerResponseDetails.cvvResponseCode) == -1 ? res.issuerResponseDetails.cvvResponseCode : 0
-        var CVVResponseDescription =  CVVResponseCode && res.issuerResponseDetails.cvvResponseDescription ? res.issuerResponseDetails.cvvResponseDescription : ''
-        //we only look for cvv not M (match), P (not processed) cvv
+    if (!res || !res.isSuccessful) {
+      this._add_internal_error("Fullsteam, Could not get token");
 
-        if(CVVResponseDescription){
+      var msg = "";
 
-          msg = "CVV Error: "+CVVResponseDescription; //takes precedence
+      if (res && res.issuerResponseDetails) {
 
-        } else {
-          //TODO check for AVS Errors, and send nicer error to buyers
-
-
-          if(this.errors_processing.length > 0 && issuerResponseCode == '00') // we already have processing error, and there isn't a issuer error, so return...
-            return false
-
-          if(msg == '' && issuerResponseDescription && issuerResponseCode != '00')
-            msg = "Error: "+issuerResponseDescription
-
-          if(msg == '' && (!issuerResponseCode || issuerResponseCode == '00'))
-            msg = "CPE2: Missing error code"
-
-          if(msg == '')
-            msg = "CPE3: Unknown issuer error"
+        var issuerResponseCode = res.issuerResponseDetails.issuerResponseCode || 0;
+        var issuerResponseDescription = res.issuerResponseDetails.issuerResponseDescription || "";
+        var avsResponseCode = res.issuerResponseDetails.avsResponseCode || "";
+        var avsResponseDescription = res.issuerResponseDetails.avsResponseDescription || "";
+        var CVVResponseCode = res.issuerResponseDetails.cvvResponseCode || "";
+        var CVVResponseDescription = res.issuerResponseDetails.cvvResponseDescription || "";
+        var responseError = res.responseCode || "";
```

Comment:
remove unused variables
```suggestion
        var avsResponseCode = res.issuerResponseDetails.avsResponseCode || "";
        var CVVResponseCode = res.issuerResponseDetails.cvvResponseCode || "";
        var responseError = res.responseCode || "";
```